apps/browser/main.cpp

```diff
@@ -54,6 +55,11 @@ static QObject *faviconmanager_factory(QQmlEngine *, QJSEngine *)
 {
     return FaviconManager::instance();
 }
+
+static QObject *bookmarkmanager_factory(QQmlEngine *, QJSEngine *)
+{
+    return BookmarkManager::instance();
```

Comment:
Fingers cross that this won't blow up durng browser shutdown. Afair Qml engine takes ownership of this but that could well be harmless if engine destroys this as BookmarkManager as only default destructor.

To avoid potential dangling pointer, you could add small fix for the BookmarkManager::instance() and change the singleton to be QPointer.